commit 7910aadb540f3710f52fe3d00a65c82efafbb306
Author:     Yuan-Ting Hsieh (謝沅廷) <yuantinghsieh@gmail.com>
AuthorDate: Tue Apr 19 14:48:27 2022 -0700
Commit:     GitHub <noreply@github.com>
CommitDate: Tue Apr 19 14:48:27 2022 -0700

    Update CI/CD requirements and tests (#376)
    
    * Update CI/CD requirements and tests
    
    * Fix main ci (#377)
    
    * Fix build doc process (#387)
    
    * Fix app testing pt model validator
    
    * Update tests
    
    * Fix format issues
    
    * Fix integration tests
    
    * Update premerge.yml
    
    * Fix format
    
    * Fix comments
---
 nvflare/private/fed/server/job_runner.py    |  3 +--
 nvflare/private/fed/server/server_engine.py | 22 ++++++++++++----------
 2 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/nvflare/private/fed/server/job_runner.py b/nvflare/private/fed/server/job_runner.py
index ef9fdc6f..868c7bb0 100644
--- a/nvflare/private/fed/server/job_runner.py
+++ b/nvflare/private/fed/server/job_runner.py
@@ -197,8 +197,7 @@ class JobRunner(FLComponent):
 
     def restore_running_job(self, run_number: str, job_id: str, job_clients, snapshot, fl_ctx: FLContext):
         engine = fl_ctx.get_engine()
-        engine.start_app_on_server(run_number, job_id=job_id,
-                                   job_clients=job_clients, snapshot=snapshot)
+        engine.start_app_on_server(run_number, job_id=job_id, job_clients=job_clients, snapshot=snapshot)
 
         try:
             job_manager = engine.get_component(SystemComponents.JOB_MANAGER)
diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index ebeea6ca..1199df49 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -169,7 +169,7 @@ class ServerEngine(ServerEngineInternalSpec):
         return ""
 
     def get_clients(self) -> [Client]:
-        return list(self.client_manager.clients.values())
+        return list(self.client_manager.get_clients().values())
 
     def validate_clients(self, client_names: List[str]) -> Tuple[List[Client], List[str]]:
         return self._get_all_clients_from_inputs(client_names)
@@ -189,8 +189,9 @@ class ServerEngine(ServerEngineInternalSpec):
                 app_custom_folder = os.path.join(app_root, "custom")
 
             open_ports = get_open_ports(2)
-            self._start_runner_process(self.args, app_root, run_number, app_custom_folder,
-                                       open_ports, job_id, job_clients, snapshot)
+            self._start_runner_process(
+                self.args, app_root, run_number, app_custom_folder, open_ports, job_id, job_clients, snapshot
+            )
 
             threading.Thread(target=self._listen_command, args=(open_ports[0], run_number)).start()
 
@@ -212,8 +213,7 @@ class ServerEngine(ServerEngineInternalSpec):
                     data = received_data.get(ServerCommandKey.DATA)
 
                     if command == ServerCommandNames.GET_CLIENTS:
-                        return_data = {ServerCommandKey.CLIENTS: clients,
-                                       ServerCommandKey.JOB_ID: job_id}
+                        return_data = {ServerCommandKey.CLIENTS: clients, ServerCommandKey.JOB_ID: job_id}
                         conn.send(return_data)
                     elif command == ServerCommandNames.AUX_SEND:
                         targets = data.get("targets")
@@ -243,8 +243,9 @@ class ServerEngine(ServerEngineInternalSpec):
                 self.engine_info.status = MachineStatus.STOPPED
                 break
 
-    def _start_runner_process(self, args, app_root, run_number, app_custom_folder, open_ports,
-                              job_id, job_clients, snapshot):
+    def _start_runner_process(
+        self, args, app_root, run_number, app_custom_folder, open_ports, job_id, job_clients, snapshot
+    ):
         new_env = os.environ.copy()
         if app_custom_folder != "":
             new_env["PYTHONPATH"] = new_env["PYTHONPATH"] + ":" + app_custom_folder
@@ -640,9 +641,10 @@ class ServerEngine(ServerEngineInternalSpec):
                     fl_ctx.set_prop(FLContextKey.JOB_INFO, (job_id, job_clients))
             else:
                 (job_id, job_clients) = job_info
-            snapshot.set_component_snapshot(component_id=SnapshotKey.JOB_INFO,
-                                            component_state={SnapshotKey.JOB_CLIENTS: job_clients,
-                                                             SnapshotKey.JOB_ID: job_id})
+            snapshot.set_component_snapshot(
+                component_id=SnapshotKey.JOB_INFO,
+                component_state={SnapshotKey.JOB_CLIENTS: job_clients, SnapshotKey.JOB_ID: job_id},
+            )
 
             snapshot.completed = completed
 
